src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3379,12 +3374,40 @@ public override void VisitNamedType(NamedTypeSymbol symbol)
             }
         }
 
+        /// <returns><see langword="true"/> if file types are present in files with duplicate file paths. Otherwise, <see langword="false" />.</returns>
         private bool CheckDuplicateFilePaths(DiagnosticBag diagnostics)
         {
             var visitor = new DuplicateFilePathsVisitor(diagnostics);
             return visitor.CheckDuplicateFilePathsAndFree(SyntaxTrees, GlobalNamespace);
         }
 
+        /// <returns><see langword="true"/> if duplicate interceptors are present in the compilation. Otherwise, <see langword="false" />.</returns>
+        private bool CheckDuplicateInterceptions(DiagnosticBag diagnostics)
```

Comment:
I am wondering if it would make sense to call this in `SourceModuleSymbol.ForceComplete`, for example, after we find `this.GlobalNamespace.HasComplete(CompletionPart.MembersCompleted)`. At this point I think we should have reported declaration diagnostics for all members, and we would know if there are duplicate interceptors for a location.